qiskit/ignis/mitigation/measurement/fitters.py

```diff
@@ -169,3 +172,148 @@ def plot_calibration(self, ax=None, show_plot=True):
+            if self.nqubits != len(state_labels[0]):
+                raise ValueError("mit_pattern does not match state_labels")
+
+        if self._results is not None:
+            self._build_calibration_matrices()
+
+    @property
+    def cal_matrices(self):
+        """Return cal_matrices."""
+        return self._cal_matrices
+
+    @cal_matrices.setter
+    def cal_matrices(self, new_cal_matrices):
+        """set cal_matrices."""
+        self._cal_matrices = copy.deepcopy(new_cal_matrices)
+
+    @property
+    def filter(self):
+        """return a measurement filter using the cal matrices"""
+        return TensoredFilter(self._cal_matrices, self._qubit_list_sizes)
+
+    @property
+    def nqubits(self):
+        """Return _qubit_list_sizes"""
+        return sum(self._qubit_list_sizes)
+
+    def readout_fidelity(self, cal_index=0, label_list=None):
+        """
+        Based on the results output the readout fidelity, which is the average
+        of the diagonal entries in the calibration matrices
+
+        Args:
+            cal_index: readout fidelity of which sub cal?
+            label_list (list of lists on states):
+            Returns the average fidelity over of the groups of states.
+            If None then each state used in the construction of the
+            calibration matrices forms a group of size 1
+
+        Returns:
+            readout fidelity (assignment fidelity)
+
+
+        Additional Information:
+            The on-diagonal elements of the calibration matrices are the
+            probabilities of measuring state 'x' given preparation of state
+            'x'
+        """
+
+        if self._cal_matrices is None:
+            raise QiskitError("Cal matrix has not been set")
+
+        if label_list is None:
+            # TODO: consider changing the default label_list,
+            # probably to all the states
+            label_list = [[label] for label in
+                          count_keys(self._qubit_list_sizes[cal_index])]
+
+        tmp_fitter = CompleteMeasFitter(None, count_keys(
+            self._qubit_list_sizes[cal_index]), circlabel='')
+        tmp_fitter.cal_matrix = self.cal_matrices[cal_index]
+        return tmp_fitter.readout_fidelity(label_list)
+
+    def _build_calibration_matrices(self):
+        """
+        Build the measurement calibration matrices from the results of running
+        the circuits returned by `measurement_calibration`
+        """
+
+        self._cal_matrices = []
+        for list_size in self._qubit_list_sizes:
+            self._cal_matrices.append(np.zeros([2**list_size, 2**list_size],
+                                               dtype=float))
+
+        for state in self._state_labels:
+            state_cnts = self._results.get_counts('%scal_%s' %
+                                                  (self._circlabel, state))
+            for measured_state, counts in state_cnts.items():
```

Comment:
This is enforcing a specific ordering on the sub matrices, maybe it should make a sub state labels?